React/todo/src/components/Item.jsx

```diff
@@ -5,37 +5,59 @@ function Item({ item, setTodos }) {
   const [editing, setEditing] = React.useState(false);
   const inputRef = React.useRef(null);
 
-  const toggleComplete = () => {
-    setTodos(prev =>
-      prev.map(t => (t.id === item.id ? { ...t, is_completed: !t.is_completed } : t))
-    );
-  };
+  function updateTodoById(id, updater) {
+    setTodos(prev => {
+      return prev.map(todo => {
+        if (todo.id !== id) return todo;
+        return updater(todo);
```

Comment:
Let's not forget about one liner if blocks and newline between conditionals, statements and returns
```suggestion
        if (todo.id !== id) {
          return todo
        };
        
        return updater(todo);
```